### ISSUE TITLE
Removing settings button or toggling it off.

### ISSUE THREAD

**Issue by Nihim** [benkaiser/node-music-player#71]
Hello,

I am thinking of using the player but since it will be accessed by several friends I wouldn't like for it to be possible to change the music directory. So it would be nice if the settings button could be hidden.

Cheers

**Comment by benkaiser** [comment]
You could strip out the html displaying the button as a quick hack if you
wanted... Otherwise yeah it would be nice to configure this from a config
file somehow. I'll keep this issue here until its added. PRs are welcome
also :P

**Comment by RolfKoenders** [comment]
Or we could think of optional user-authentication (roles etc.)?

**Comment by benkaiser** [comment]
Yeah that's the optimal implementation of it I think @RolfKoenders

**Comment by Nihim** [comment]
I am afraid I can't help anyhow since even installing it with node requires quite some reading for me :)
Though I have to say I dislike players that have admin / user logins etc and end up having too many features. Some kind of config for some "hard rules" would be nice, though in the end it's your player you decide what's the best choice!

Also is there a way to enable https to the player outside using a proxy from the webserver?

**Comment by benkaiser** [comment]
No worries :)

Yeah I'm still not to sure. I would definitely want to keep the current
functionality default and just allow some hard config rules (so they can't
be enabled via web interface) to enable authentication and such. But I'm
still not certain about how to do that yet / have the time to implement it.

Your best bet is putting it behind a proxy. I would recommend nginx, as it
allows you to port forward from 80 (or 443 for https) to whatever port you
have it running on, along with using certificates along the way. If you get
stuck with this I have some nginx configs (not https, but there's others on
the internet for that) that I could share with you if you'd like.

**Comment by Nihim** [comment]
It's cool :)

Yea seems will have to try the proxy approach. I only got experience with lighttpd and from asking friends that are well "proficient" in nix it can be done just fine with lighttpd, I just have to find out how!

Just, I also, need to find time to play around a bit.

**Comment by benkaiser** [comment]
All good. Best of luck with it.

**Comment by ohnx** [comment]
Speaking of reverse proxying, I seem to have some issues with the web
client connecting to socket.io.

I set up nginx to forward all requests to the correct port. The web client
loads, but I get an error saying that the socket failed to connect.
However, after a bit, it ends up connecting and working.

Is this an issue with my nginx configuration?

**Comment by benkaiser** [comment]
Yeah I've noticed it's pretty flaky too. I need to look into it.

**Comment by benkaiser** [comment]
Okay, so I've made a new branch called `no_ui_settings`. I recommend you:

1. use master to setup the server, change music directory, rescan library, etc
2. change to `no_ui_settings` branch and let people access that

I've changed the public demo to use the `no_ui_settings` version. It simply changes the view, if someone mucked around in the console and fired the changes themselves it would work just fine. The code is open source after all.

This works for a basic implementation satisfying this feature request for now. I'll will add proper authentication options down the line.
